lib/web/apps.go

```diff
@@ -146,21 +146,28 @@ func (h *Handler) clusterAppsGet(w http.ResponseWriter, r *http.Request, p httpr
 	}, nil
 }
 
-type GetAppFQDNRequest ResolveAppParams
+type GetAppDetailsRequest ResolveAppParams
 
-type GetAppFQDNResponse struct {
+type GetAppDetailsResponse struct {
 	// FQDN is application FQDN.
 	FQDN string `json:"fqdn"`
+	// RequiredApps is a list of required app names
+	RequiredApps []string `json:"requiredApps"`
 }
 
-// getAppFQDN resolves the input params to a known application and returns
-// its valid FQDN.
+// getAppDetails resolves the input params to a known application and returns
+// its app details.
 //
 // GET /v1/webapi/apps/:fqdnHint/:clusterName/:publicAddr
-func (h *Handler) getAppFQDN(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext) (interface{}, error) {
-	req := GetAppFQDNRequest{
+func (h *Handler) getAppDetails(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext) (interface{}, error) {
```

Comment:
suggestion: If we can send required apps (fqdn) in the app response from the unified resource api, we can collapse the first two requests we have now to one.  
1. First request `/web/launch...?`
2. Second request `webapi/apps` responds with required apps
3. Third request `/web/launch...?required-apps=<required app from responds received in second request >`.

I can see this URL was always called to get the app fqdn which I do not understand why it was done that way. Up to you if you want to reduce number of requests here.

Reply:
That or looks like we cna also combine the state and required apps response to one, which would reduce few extra requests.

Reply:
Agree, but I will TODO this for now. I'd like to talk to @capnspacehook to make sure this extra call can be merged with the first. Good idea!